src/main/java/io/vertx/core/http/impl/HttpUtils.java

```diff
@@ -424,7 +424,15 @@ public static URI resolveURIReference(URI base, String ref) throws URISyntaxExce
   /**
    * Extract the path out of the uri.
    */
-  static String parsePath(String uri) {
+  static String parsePath(final String uri) {
+    final Object o = parsePathAndQueryStartIf(uri);
+    return (String) ( o instanceof Object[] ? ((Object[])o)[0] : o );
+  }
+
+  /**
+   * Extract the path out of the uri and queryStart
+   */
+  static Object parsePathAndQueryStartIf(final String uri) {
```

Comment:
I think this is a bad name, I can't tell what this method does from it's name

Reply:
Any suggesion  for name ?